### ISSUE TITLE
about "greg_loss", "predictor_greg_loss", "biases_greg_loss"

### ISSUE THREAD

**Issue by Neronjust2017** [DMIRLAB-Group/DSR#1]
Hi,
Thanks for your code! However, There's something I don't quite understand. What are "weight_para", "bias_para", "label_b_para" and "greg_loss", "biases_greg_loss", "predictor_greg_loss" for? It  seems that these terms are not be mentioned in the paper. Could you please explain for that? Thanks!

**Comment by jozerozero** [comment]
Hi, thanks for focusing on our works. Let's clarify the details of the codes:
(1) "weight_para", "bias_para", "label_b_para" denote different parameters of our code, this is because different parameters should be updated in different learning rate. If 
(2) "*_greg_loss" denotes the l2 restriction on different parameters.

We will upload the latest version to the Axriv later.
Thanks a lot.

**Comment by Neronjust2017** [comment]
Hi, 
Thanks for your explanation. I have two other questions right now, hope you can answer that for me.
(1)  In "office_home_sdr.py",
![图片](https://user-images.githubusercontent.com/27863360/101436696-a1825080-3949-11eb-961c-0d45b50871a6.png)

Without GRL, optimizing this loss means maximizing the entropy, because "src_entropy and tgt_entropy is the negative entropy".
With GRL, optimizing this loss means minimizing the entropy. 
![图片](https://user-images.githubusercontent.com/27863360/101436732-b1019980-3949-11eb-8f58-7dd797fec95d.png)

Is there someting wrong here? 

(2) I add some tensorboard log code in "office_home_dsr.py" without modifying the model and training, and I plot some metric curves. 
![图片](https://user-images.githubusercontent.com/27863360/101435334-2750cc80-3947-11eb-89e8-3675aaf27e1e.png)

![图片](https://user-images.githubusercontent.com/27863360/101434552-b78e1200-3945-11eb-836c-cf98836352f2.png)
It seems In the whole training process, "domain_ domain_ acc" and "semantic_ domain_acc" do not improve, because the average values of source and target are close to 0.5, which indicates that semantic latent variables and domain latent variables do not help to distinguish which domain the sample comes from (I konw semantic latent variables cannot distinguish the domain of the sample is an optimization goal). This means that domain latent variables (right half part of your model)do not work. Why is this?

Thanks!

**Comment by JinProton** [comment]
Hi,
Thanks for your code! However, Is there any part of your code being used for adversarial?

Thanks!
